### PR TITLE
Remove dependency on jdk.compiler from builtin method processor

### LINKED ISSUE #5817: Avoid using Javac internals in Enso DSL
When trying to upgrade to new version of Frgaal I noticed compilation problems in `lib/scala/interpreter-dsl` project. It is using internals of `com.sun.tools.javac` package which is not official JDK API. It'd be better to avoid accessing internals and use proper API - which probably is [AnnotationValueVisitor](https://docs.oracle.com/javase/8/docs/api/javax/lang/model/util/SimpleAnnotationValueVisitor8.html).

Following diff shows the affected code:
```diff
diff --git lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java
index 6c491df75..dc540a81a 100644
--- lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java
+++ lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java
@@ -1,23 +1,22 @@
 package org.enso.interpreter.dsl.builtins;
 
-import com.google.common.base.CaseFormat;
-import com.sun.tools.javac.code.Attribute;
-import com.sun.tools.javac.code.Symbol;
-import com.sun.tools.javac.util.Pair;
-import org.apache.commons.lang3.StringUtils;
-import org.enso.interpreter.dsl.Builtin;
-
-import javax.annotation.processing.ProcessingEnvironment;
-import javax.lang.model.element.*;
-import javax.lang.model.type.TypeMirror;
-import javax.lang.model.util.Types;
 import java.lang.annotation.Annotation;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Types;
+
+import org.apache.commons.lang3.StringUtils;
+import org.enso.interpreter.dsl.Builtin;
+
+import com.google.common.base.CaseFormat;
+
 public abstract class MethodGenerator {
   protected final boolean isStatic;
   protected final boolean isConstructor;
@@ -180,24 +179,6 @@ public abstract class MethodGenerator {
       TypeMirror builtinType = builtinElement.asType();
 
       List<SafeWrapException> exceptionWrappers = new ArrayList<>();
-      for (AnnotationMirror am : element.getAnnotationMirrors()) {
-        if (am.getAnnotationType().equals(builtinType)) {
-          Attribute.Class valueFrom = null;
-          Attribute.Class valueTo = null;
-          for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> entry :
-              am.getElementValues().entrySet()) {
-            Name key = entry.getKey().getSimpleName();
-            if (key.toString().equals(FromElementName)) {
-              valueFrom = (Attribute.Class) (entry.getValue());
-            } else if (key.toString().equals(ToElementName)) {
-              valueTo = (Attribute.Class) (entry.getValue());
-            }
-          }
-          if (valueFrom != null) {
-            exceptionWrappers.add(new SafeWrapException(valueFrom, Optional.ofNullable(valueTo)));
-          }
-        }
-      }
       return exceptionWrappers.toArray(new SafeWrapException[0]);
     }
 
@@ -210,35 +191,6 @@ public abstract class MethodGenerator {
       TypeMirror builtinType = builtinElement.asType();
 
       List<SafeWrapException> wrappedExceptions = new ArrayList<>();
-      for (AnnotationMirror am : element.getAnnotationMirrors()) {
-        if (tpeUtils.isSameType(am.getAnnotationType(), builtinType)) {
-          for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> entry :
-              am.getElementValues().entrySet()) {
-            if (ValueElementName.equals(entry.getKey().getSimpleName().toString())) {
-              Attribute.Array wrapExceptions = (Attribute.Array) entry.getValue();
-              for (int i = 0; i < wrapExceptions.values.length; i++) {
-                Attribute.Class valueFrom = null;
-                Attribute.Class valueTo = null;
-                Attribute.Compound attr = (Attribute.Compound) wrapExceptions.values[i];
-                for (Pair<Symbol.MethodSymbol, Attribute> p : attr.values) {
-                  Name key = p.fst.getSimpleName();
-                  if (key.contentEquals(FromElementName)) {
-                    valueFrom = (Attribute.Class) p.snd;
-                  } else if (key.contentEquals(ToElementName)) {
-                    valueTo = (Attribute.Class) p.snd;
-                  }
-                }
-                if (valueFrom != null) {
-                  SafeWrapException converted =
-                      new SafeWrapException(valueFrom, Optional.ofNullable(valueTo));
-                  wrappedExceptions.add(converted);
-                }
-              }
-              break;
-            }
-          }
-        }
-      }
       return wrappedExceptions.toArray(new SafeWrapException[0]);
     }
   }
diff --git lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/SafeWrapException.java lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/SafeWrapException.java
index 9d98a609a..cf521cac5 100644
--- lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/SafeWrapException.java
+++ lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/SafeWrapException.java
@@ -1,19 +1,15 @@
 package org.enso.interpreter.dsl.builtins;
 
-import com.sun.tools.javac.code.Attribute;
-import org.apache.commons.lang3.StringUtils;
-import org.enso.interpreter.dsl.Builtin;
-
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
-import java.util.stream.Collectors;
+
+import org.enso.interpreter.dsl.Builtin;
 
 /**
  * Wrapper around {@link Builtin.WrapException} annotation with all elements of Class type resolved.
  * extracted
  */
-public record SafeWrapException(Attribute.Class from, Optional<Attribute.Class> to) {
+public record SafeWrapException(Object from, Optional<Object> to) {
 
     private static final String PanicExceptionClassName = "PanicException";
     private static final String UnsupportedMessageExceptionClassName = "UnsupportedMessageException";
@@ -54,8 +50,8 @@ public record SafeWrapException(Attribute.Class from, Optional<Attribute.Class>
         return clazz.map(c -> c.equals(PanicExceptionClassName)).orElse(true);
     }
 
-    private String fromAttributeToClassName(Attribute.Class clazz, Boolean fullName) {
-        String baseType = clazz.classType.baseType().toString();
+    private String fromAttributeToClassName(Object clazz, Boolean fullName) {
+        String baseType = clazz.toString();
         if (fullName) return baseType;
         else {
             String[] clazzElements = baseType.split("\\.");
```

- [] (Optionally) Make the build fail when using javac internals
- [] Rewrite the "safe exception" extraction to use official JDK API

### PR DESCRIPTION
**Pull request by Akirathan**
Closes #5817

### Pull Request Description

Remove the dependency on `jdk.compiler` in `interpreter-dsl`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java [hunk 1]
```diff
@@ -144,121 +142,113 @@ protected MethodParameter fromVariableElementToMethodParameter(
...
+        elements.add(safeWrapException);
+      }
+      return elements;
+    }
+  }
+
+  private class AnnotationTypeVisitor extends SimpleAnnotationValueVisitor14<TypeElement, Object> {
+    @Override
+    public TypeElement visitType(TypeMirror t, Object o) {
+      var typeElement = processingEnvironment.getElementUtils().getTypeElement(t.toString());
```
Comment (JaroslavTulach) [inline lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java]: There should be a way to convert `TypeMirror` to element. I think it is: [Types.asElement](https://docs.oracle.com/javase/8/docs/api/javax/lang/model/util/Types.html#asElement-javax.lang.model.type.TypeMirror-)

**Review by JaroslavTulach — APPROVED** [review]
Way nicer than our yesterday's prototype! Just avoid `typeMirror.toString()` and it is going to be beautiful!

**Review by hubertp — APPROVED** [review]

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java (hunk 1 above)
Comment (Akirathan) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java]: What is wrong with the current way? But your proposal converts only `TypeMirror` to `Element` and not to `TypeElement`.

Comment (Akirathan) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java]: Turns out there is a way to do that with [ElementVisitor](https://docs.oracle.com/javase/8/docs/api/javax/lang/model/element/ElementVisitor.html).

Comment (Akirathan) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java]: Fixed in 1cf518556b73a6fa082625844dcafb78b92a558f

Comment (JaroslavTulach) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java]: Visitor over visitor, but that's how the JDK guys want it to be! No more complains, this is the nicest annotation processor code I can imagine.

**Review by 4e6 — APPROVED** [review]

**Pull request merged**